flow/Trace.cpp

```diff
@@ -1442,3 +1475,27 @@ std::string traceableStringToString(const char* value, size_t S) {
 
 	return std::string(value, S - 1); // Exclude trailing \0 byte
 }
+
+RealTimePrinter::RealTimePrinter() : beginTime(Clock::now()), flowBeginTime(g_network->now()) {}
+
+// converts the given flow time into a
+// string. now has to be larger
+// then flowBeginTime.
+// %Y-%m-%dT%H:%M:%S
+// This only has second-resolution for the simple reason
+// that std::put_time does not support higher resolution.
+// As we always also log the flow time and this is only for
+// human readability, this is good enough for now.
+std::string RealTimePrinter::toString(double now) {
+	time_t n = Clock::to_time_t(beginTime + std::chrono::microseconds(std::lround(1e6 * (now - flowBeginTime))));
+	if (g_network->isSimulated()) {
+		// in simulation, the clock is simulated as well. Therefore,
+		// actual time might be very different from simulated time.
+		n = Clock::to_time_t(Clock::now());
+	}
+	std::stringstream ss;
+	// Use thread-safe localtime_r instead of localtime
+	struct tm result;
+	ss << std::put_time(::localtime_r(&n, &result), "%Y-%m-%dT%H:%M:%S");
```

Comment:
Is this going to be expressed in local time or UTC?

Reply:
utc

Reply:
I just tried running this in a standalone example, and I seem to be getting times printed in my local timezone (PDT). Using `gmtime` instead of `localtime` seems to give me the correct result.

Also, one suggestion that I thought was good when we discussed this was to include the `Z` suffix to denote that it is in UTC:

`1970-01-01T00:00:00Z`

Reply:
Thanks AJ!  Will make both of these changes